### ISSUE TITLE
Ray geometry artifact running simple AR demo on WebXR Viewer by Mozilla

### ISSUE THREAD

**Issue by AdamRamberg** [react-spring/react-xr#3]
Hi there! 

First of all, thanks for an amazing library! I've been fiddling around with the [AR demo](https://codesandbox.io/s/react-xr-simple-ar-demo-8w8hm) on [WebXR Viewer](https://apps.apple.com/us/app/webxr-viewer/id1295998056) by Mozilla (iOS). When running the demo I've noticed a meshbeing rendered that is covering a large portion of the screen, which occurs when the `Select` and `Hover` components are being used on a raycast hit. Digging into the code base I realized that this was due to that the raycast mesh (rendered once per controller by `DefaultXRControllers`) is too close to the camera when visible. See [this](https://github.com/react-spring/react-xr/blob/67335ddf74e62f1dff70a349933f52a7a8b0024f/src/XR.tsx#L198) line, which places the mesh's closest side at the same position as the camera. By adding an offset to the mesh moving it away from the camera the artifact is gone, eg: 

```
      const OFFSET = 0.25
      ray.visible = true
      ray.scale.y = it.hoverRayLength - OFFSET / 2
      ray.position.z = -it.hoverRayLength / 2 - OFFSET
``` 

I'm quite new to three.js and WebXR so I'm not 100% sure if this is the correct approach of solving this problem. Another approach might for example be to set some clipping option on the camera. 

Futhermore, the artifact is not present on Android and Chrome (latest release). 

Let me know if you need any further info. Wouldn't mind creating a PR for whatever solution we come up with.

**Comment by sniok** [comment]
Hi! 
Thanks for opening the issue, this is very interesting as I'd expect there would be no ray line in the AR demo because usually (well on android) controller should have [handedness](https://immersive-web.github.io/webxr/#enumdef-xrhandedness) set to 'none' and that case is handled [here](https://github.com/react-spring/react-xr/blob/67335ddf74e62f1dff70a349933f52a7a8b0024f/src/XR.tsx#L191). Offsetting the line could work but we would also need to test it with regular VR controllers, so that ray doesn't start outside controllers.

another solution could be adjusting [near](https://threejs.org/docs/#api/en/cameras/PerspectiveCamera.near) value on the threejs camera. or we could disable that line altogether in AR mode.

**Comment by sniok** [comment]
quick update: I managed to reproduce the artifact and it appears that XRInputSourcesource of WebXR viewer states its' handedness as "right" :D

**Comment by sniok** [comment]
Offset fix seems to work. Tested it on android + chrome, iOS + webxr viewer and in oculus browser on quest. Released in 1.0.8

**Comment by AdamRamberg** [comment]
Thanks for the quick fix! ⭐
